File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int j =0, ans = 0;
        int k[300];
        if (s=="") return 0;
        for (int i =0; i<300; i++){
            k[i] = -1;
        }
        int i =0;
        for(int j =0; j<s.length(); j++){
            // if (j == s.length() -1){

            // }
            // cout<<k[s[j]];
            if (k[s[j]] ==-1){
                k[s[j]] = j;
            }
            else{
                // cout<<j<<" "<<i<<" "<<k[s[j]]<<endl;
                if (ans ==0) ans = j-i;
                // ans = max(ans, min(j-i, j-k[s[j]]));
                i = max(i,k[s[j]] +1);
                k[s[j]] = j;
            }
            cout<<j<<" "<<i<<endl;
            ans = max(ans, j-i+1);
        }

        // j = s.length();
        // ans = max(ans, j-i+1);    
        if (s.length() && !ans) return s.length();
        return ans;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s == "":
            return 0
        ans = 0
        left = 0
        right = 1
        count = 1
        while right<len(s):
            if s[right] in s[left:right]:
                ans = max(ans, count)
                left = s[left:right].find(s[right]) +1+left
                print(left, right)
                count = right-left +1
            else:
                count += 1
            right += 1
        ans = max(ans, count)
        return ans
        # for i in range(len(s)):
        #     if s[i] in s[:i]:
        #         ans = 
